src/main/java/org/reficio/p2/P2Mojo.java

```diff
@@ -330,7 +340,21 @@ private void processFeatures() {
         Multimap<P2Artifact, ResolvedArtifact> resolvedArtifacts = ArrayListMultimap.create();
         for (P2Artifact p2Artifact : artifacts) {
             logResolving(p2Artifact);
-            ArtifactResolutionResult resolutionResult = resolveArtifact(p2Artifact);
+            ArtifactResolutionResult resolutionResult;
+            try {
+                resolutionResult = resolveArtifact(p2Artifact);
+            } catch (final Exception ex) {
+                // In fact a org.eclipse.aether.resolution.ArtifactResolutionException will be thrown, if artifact
+                // couldn't be resolved, but this checked exception does not occur in the method signatures here.
+                // So we catch all exceptions and wrap them into a runtime exception to not change all the method
+                // signatures.
+                if (skipNotExistingArtifacts) {
+                    log.warn(String.format("Skip artifact=[%s]: %s", p2Artifact.getId(), ex.getMessage()));
+                    continue;
```

Comment:
I just copied from https://github.com/reficio/p2-maven-plugin/blob/master/src/main/java/org/reficio/p2/P2Mojo.java#L304 to be consistent with your style of logging.